Automattic/wp-calypso
# Products: Remove formatProduct from remaining jetpack functions

#### Changes proposed in this Pull Request

In the vein of https://github.com/Automattic/wp-calypso/pull/58197, https://github.com/Automattic/wp-calypso/pull/58136, #58080, and #58086, this PR removes more uses of the `formatProduct` function in the `calypso-products` package because it makes it difficult to determine types and in some cases is unnecessary.

In this PR specifically, we change the following functions and convert them to TypeScript. None of these changes modify the functionality of these functions, except that they now always require an object (where they previously would accept anything).

- `isJetpackSearch`
- `isJetpackScan`
- `isJetpackProduct`
- `isJetpackPremium`
- `isJetpackBusiness`
- `isJetpackBackup`
- `isJetpackAntiSpam`
- `isGuidedTransfer`

#### Testing instructions

No specific testing should be necessary (but see below).

One thing to keep in mind about this change is that these functions will no longer work if passed something that's not an object (eg: a boolean or a string). I believe this is a good thing because any such use is probably a bug, however, this type of change revealed [a breaking bug in signup](https://github.com/Automattic/wp-calypso/pull/58189) in one PR of its kind, so it's worth going through the call sites of the functions above and making sure that they are never called with non-object values (I've done this but feel free to check yourself).
